File: angular/src/app/expense/models/create-update-expense-dto.ts

import { ExpenseType } from './expense-type';
export class CreateUpdateExpenseDto  {
  title: string;
  onDate: string;
  type: any;
  amount: number;

  constructor(initialValues: Partial<CreateUpdateExpenseDto> = {}) {
    if (initialValues) {
      for (const key in initialValues) {
        if (initialValues.hasOwnProperty(key)) {
          this[key] = initialValues[key];
        }
      }
    }
  }
}

File: aspnet-core/src/Nic.Octopus.Application.Contracts/CreateUpdateExpenseDto.cs
using System;
using System.ComponentModel.DataAnnotations;
using Nic.Octopus.Domain.Shared;

namespace Nic.Octopus.Application.Contracts
{
    public class CreateUpdateExpenseDto
    {
        [Required]
        [StringLength(128)]
        public string Title { get; set; }

        [Required]
        public DateTime OnDate { get; set; }

        [Required]
        public ExpenseType Type { get; set; }

        [Required]
        public float Amount { get; set; }
    }
}